mmv1/products/discoveryengine/SearchEngine.yaml

```diff
@@ -0,0 +1,147 @@
+name: 'SearchEngine'
+description: |
+  Vertex AI Search and Conversation can be used to create a search engine or a chat application by connecting it with a datastore
+references: !ruby/object:Api::Resource::ReferenceLinks
+  guides:
+    'Create a Search Engine': 'https://cloud.google.com/generative-ai-app-builder/docs/create-engine-es'
+  api: 'https://cloud.google.com/generative-ai-app-builder/docs/reference/rest/v1/projects.locations.collections.engines'
+
+base_url: 'projects/{{project}}/locations/{{location}}/collections/{{collection_id}}/engines/{{engine_id}}'
+self_link: 'projects/{{project}}/locations/{{location}}/collections/{{collection_id}}/engines/{{engine_id}}'
+create_url: 'projects/{{project}}/locations/{{location}}/collections/{{collection_id}}/engines?engineId={{engine_id}}'
+delete_url: 'projects/{{project}}/locations/{{location}}/collections/{{collection_id}}/engines/{{engine_id}}'
+
+update_verb: :PATCH
+update_mask: true
+
+import_format:
+  [
+    'projects/{{project}}/locations/{{location}}/collections/{{collection_id}}/engines/{{engine_id}}',
+  ]
+
+examples:
+  - !ruby/object:Provider::Terraform::Examples
+    name: "discoveryengine_searchengine_basic"
+    primary_resource_id: 'basic'
+    primary_resource_name:
+      'fmt.Sprintf("tf_test_search_engine%s", context["random_suffix"])'
+    vars:
+      engine_id: "example-engine-id"
+      data_store_id: "example-datastore-id"
+
+parameters:
+  - !ruby/object:Api::Type::String
+    name: 'engine_id'
+    description: |
+      Unique ID to use for Search Engine App.
+    required: true
+    immutable: true
+    url_param_only: true
+  - !ruby/object:Api::Type::String
+    name: 'collection_id'
+    description: |
+      The collection ID.
+    required: true
+    immutable: true
+    url_param_only: true
+  - !ruby/object:Api::Type::String
+    name: 'location'
+    description: |
+      Location.
+    required: true
+    immutable: true
+    url_param_only: true
+
+properties:
+  - !ruby/object:Api::Type::Enum
+    name: 'industryVertical'
+    description: |
+      Industry Vertical
+    immutable: true
+    values:
+      - :GENERIC
+      - :MEDIA
+  - !ruby/object:Api::Type::String
+    name: 'displayName'
+    description: |
+      Display Name
+    required: true
+  - !ruby/object:Api::Type::Array
+    name: 'dataStoreIds'
+    description: |
+      Data Store ID
+    required: true
+    immutable: true
+    item_type: Api::Type::String
+  - !ruby/object:Api::Type::Enum
+    name: 'solutionType'
+    description: |
+      A solution type
+    immutable: true
+    values:
+      - :SOLUTION_TYPE_SEARCH
+  - !ruby/object:Api::Type::Time
+    name: 'createTime'
+    description: |
+      Timestamp the Engine was created at.
+    output: true
+  - !ruby/object:Api::Type::Time
+    name: 'updateTime'
+    description: |
+      Timestamp the Engine was last updated.
+    output: true
+  - !ruby/object:Api::Type::NestedObject
+    name: 'searchEngineConfig'
```

Comment:
Is this config not required?

Reply:
Yes this is not required. This is an advanced configuration for Search Engine
@chenlei1216 Please confirm